### PR TITLE
feature/TINY-7119: BDD conversion Legacy Output

### PR DESCRIPTION
**Pull request by camiloHimura**
Related Ticket:

Description of Changes:
* Refactor plugins to BDD-style tests

Pre-checks:
~* [ ] Changelog entry added~
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)
* [x] License headers added on new files (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts [hunk 1]
```diff
@@ -1,152 +1,176 @@
...
+    indent: false,
+    base_url: '/project/tinymce/js/tinymce',
+    font_formats: 'Arial=arial,helvetica,sans-serif;',
+    setup: (editor) => {
+      // Store the formats on `PostRender`, which is fired before the initial editor content is loaded
+      editor.on('PostRender', () => {
+        formatsCell.set({ ...editor.formatter.get() });
+      });
+    }
+  }, [ Theme ]);
```
Comment (lnewson) [inline modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts:22]: This is missing the `Plugin` import which used to exist and has incorrectly been removed by the looks? This also might be why the tests are failing on CI at a glance.

File: modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts [hunk 2]
```diff
@@ -1,152 +1,176 @@
...
+
+  it('TBA: Setting overrides', () => {
+    const editor = hook.editor();
+    assert.equal(editor.getParam('inline_styles'), false);
```
Comment (lnewson) [inline modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts:26]:
```suggestion
    assert.isFalse(editor.getParam('inline_styles'));
```

File: modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts [hunk 3]
```diff
@@ -1,152 +1,176 @@
...
+    editor.setContent('<p><b class="abc" style="color: red; font-size: 20px;" data-test="2">text</b></p>');
+    TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0, 0 ], 4);
+
+    editor.execCommand('bold');
+    TinyAssertions.assertContent(editor, '<p><span class="abc" style="color: red; font-size: 20px;">text</span></p>');
+  });
+
+  it('TBA: Formats registered before loading initial content', () => {
+    const formats = formatsCell.get();
+    assert.deepInclude(formats.bold[0], { inline: 'b', remove: 'all', deep: true, split: true, preserve_attributes: [ 'class', 'style' ] });
```
Comment (lnewson) [inline modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts:171]: Should these be `deepEqual` as that'd be equivalent to the previous version? as `deepInclude` allows other items or it to be at a different depth IIRC.

File: modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts [hunk 4]
```diff
@@ -1,152 +1,176 @@
...
-import { UnitTest } from '@ephox/bedrock-client';
+import { describe, it } from '@ephox/bedrock-client';
 import { Cell } from '@ephox/katamari';
-import { LegacyUnit, TinyLoader } from '@ephox/mcagar';
+import { TinyAssertions, TinyHooks, TinySelections } from '@ephox/mcagar';
+import { assert } from 'chai';
+import Editor from 'tinymce/core/api/Editor';
 
-import Plugin from 'tinymce/plugins/legacyoutput/Plugin';
 import Theme from 'tinymce/themes/silver/Theme';
```
Comment (lnewson) [inline modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts:7]: This should be beside the `Plugin`/`Theme` import as it's an internal import. eg:
```
import { assert } from 'chai';

import Editor from 'tinymce/core/api/Editor';
import Plugin from 'tinymce/plugins/legacyoutput/Plugin';
import Theme from 'tinymce/themes/silver/Theme';
```

File: modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts [hunk 5]
```diff
@@ -1,152 +1,176 @@
...
+    TinySelections.setSelection(editor, [ 0 ], 0, [ 0, 0 ], 4);
+
+    editor.execCommand('justifyfull');
+    TinyAssertions.assertContent(editor, '<p align="justify">text</p>');
+  });
+
+  it('TBA: Justifycenter image', () => {
+    const editor = hook.editor();
+    editor.setContent('<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAwAAAAMCAYAAABWdVznAAAAF0lEQVR42mP8/5/hPwMJgHFUw6gG7AAAXVgj6XowjMAAAAAASUVORK5CYII=" /></p>');
+    TinySelections.setSelection(editor, [ 0 ], 0, [ 0 ], 0);
```
Comment (lnewson) [inline modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts:134]:
```suggestion
    TinySelections.setCursor(editor, [ 0 ], 0);
```

**Review by lnewson — CHANGES_REQUESTED** [review]
Looks pretty good overall, so nice work! Just a few things that were lost and a few things that need cleaning up :+1:

File: modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts (hunk 1 above)
Comment (camiloHimura) [reply modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts]: thanks to pointing it out, updated

File: modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts (hunk 2 above)
Comment (camiloHimura) [reply modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts]: updated

File: modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts (hunk 4 above)
Comment (camiloHimura) [reply modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts]: updated

File: modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts (hunk 5 above)
Comment (camiloHimura) [reply modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts]: updated

File: modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts [hunk 6]
```diff
@@ -1,152 +1,177 @@
...
+  it('TBA: Setting overrides', () => {
+    const editor = hook.editor();
+    assert.isFalse(editor.getParam('inline_styles'));
+    assert.equal(editor.getParam('fontsize_formats'), '8pt=1 10pt=2 12pt=3 14pt=4 18pt=5 24pt=6 36pt=7');
+    assert.equal(editor.getParam('font_formats'), 'Arial=arial,helvetica,sans-serif;');
+  });
+
+  it('TBA: Font color', () => {
+    const editor = hook.editor();
+    editor.focus();
```
Comment (lnewson) [inline modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts:34]: I missed this earlier sorry, but this probably shouldn't be here and instead, it should be set as the last argument in the `TinyHooks.bddSetupLight` call.
```suggestion
```

File: modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts [hunk 7]
```diff
@@ -1,152 +1,177 @@
...
+    indent: false,
+    base_url: '/project/tinymce/js/tinymce',
+    font_formats: 'Arial=arial,helvetica,sans-serif;',
+    setup: (editor) => {
+      // Store the formats on `PostRender`, which is fired before the initial editor content is loaded
+      editor.on('PostRender', () => {
+        formatsCell.set({ ...editor.formatter.get() });
+      });
+    }
+  }, [ Plugin, Theme ]);
```
Comment (lnewson) [inline modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts:23]:
```suggestion
  }, [ Plugin, Theme ], true);
```

This makes the editor focus once it's setup, which is effectively the same as what the `editor.focus()` call was trying to do below.

**Review by lnewson — APPROVED** [review]

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts (hunk 6 above)
Comment (camiloHimura) [reply modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts]: done

File: modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts (hunk 7 above)
Comment (camiloHimura) [reply modules/tinymce/src/plugins/legacyoutput/test/ts/browser/LegacyOutputPluginTest.ts]: done

**Pull request merged**